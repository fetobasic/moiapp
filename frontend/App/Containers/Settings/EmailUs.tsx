import React, { useEffect, useMemo, useRef, useState } from 'react';
import { View, StyleSheet, TextInput as TextInputRN, ScrollView, Keyboard } from 'react-native';
import { ApplicationStyles, Metrics } from 'App/Themes';
import { Button, CustomPickerSelect, HeaderSimple as Header, TextInput } from 'App/Components';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { HelpStackParamList } from 'App/Types/NavigationStackParamList';
import ValidationErrorList from 'App/Components/ValidationErrorList';
import { validate } from 'App/Services/Validator';
import { emailUsRules, textAreaLength } from './validateRules';
import { getSubjectData, AdditionalSubjectType } from 'App/Config/FeedbackFormTypes';
import { useAppSelector, useEvents, useMount } from 'App/Hooks';
import { applicationActions } from 'App/Store/Application';
import { useDispatch } from 'react-redux';
import { SubjectType } from 'App/Types/FeedbackForm';
import { modalInfo, showSnackbarMessage } from 'App/Services/Alert';
import AppConfig from 'App/Config/AppConfig';
import { TextInputRef } from 'App/Components/TextInput';
import { useKeyboard } from '@react-native-community/hooks';

type Props = NativeStackScreenProps<HelpStackParamList, 'EmailUs'>;
type EmailUs = {
  subject: string;
  email: string;
  phoneNumber: string;
  message: string;
  name: string;
};

const EmailUs = ({ route, navigation }: Props) => {
  const { yeti, firmwareVersionFailed, thingName, feedbackSubject } = route.params || {};
  const dispatch = useDispatch();
  const { keyboardShown, keyboardHeight } = useKeyboard();
  const { track } = useEvents();

  const emailInputRef = useRef<TextInputRN>();
  const nameInputRef = useRef<TextInputRN>();
  const phoneNumberInputRef = useRef<TextInputRN>();
  const messageInputRef = useRef<TextInputRN>();
  const isTracking = useRef(false);

  const { feedbackFormInfo, userEmail } = useAppSelector((state) => ({
    feedbackFormInfo: state.application.feedbackFormInfo,
    userEmail: state.auth?.user?.email || '',
  }));

  const [{ subject, email, phoneNumber, message, name }, setInput] = useState<EmailUs>({
    subject: feedbackSubject || '',
    email: userEmail,
    phoneNumber: '',
    message: '',
    name: '',
  });

  const [validationErrors, setValidationErrors] = useState<{ [field: string]: string[] }>({
    subject: [],
    email: [],
    phoneNumber: [],
    message: [],
    name: [],
  });

  useEffect(() => {
    if (feedbackFormInfo?.isSuccess) {
      if (!isTracking.current) {
        isTracking.current = true;
        track('feedback_sent', {
          subject,
          email,
          name,
          phoneNumber,
          thingName,
        });
      }

      modalInfo({
        title: 'Thank You!',
        body: 'Your message has been sent.',
        type: 'INFO',
        buttons: [{ title: 'OK', action: () => navigation.goBack() }],
      });
      setInput({ subject: '', email: '', phoneNumber: '', message: '', name: '' });
      Keyboard.dismiss();
    }

    if (feedbackFormInfo?.isError) {
      showSnackbarMessage("Can't send feedback. Try again later.", true);
      Keyboard.dismiss();
    }
  }, [email, feedbackFormInfo, navigation, phoneNumber, name, subject, thingName, track]);

  useMount(() => {
    return () => {
      dispatch(applicationActions.clearFeedbackFormInfo());
    };
  });

  const onChangeHandler = (value: string, keyName: keyof EmailUs) =>
    setInput((prevState) => ({ ...prevState, [keyName]: value }));

  const onSendHandler = () => {
    const { isValid, errors } = validate({ subject, email, name, phoneNumber, message }, emailUsRules);

    setValidationErrors(errors);

    if (!isValid) {
      Keyboard.dismiss();
    }

    if (isValid) {
      dispatch(
        applicationActions.sendFeedbackForm.request({
          email,
          message,
          name,
          phoneNumber,
          subject: subject as SubjectType,
          yeti,
          firmwareVersionFailed,
          thingName,
        }),
      );
    }
  };

  const isBtnDisabled: boolean = useMemo(
    () => Boolean(feedbackFormInfo?.isSuccess || feedbackFormInfo?.isSending),
    [feedbackFormInfo?.isSending, feedbackFormInfo?.isSuccess],
  );

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header title="Email Us" />
      <ScrollView
        style={styles.container}
        keyboardShouldPersistTaps="handled"
        contentContainerStyle={keyboardShown && { paddingBottom: keyboardHeight + 30 }}>
        <View style={styles.selectContainer}>
          <View style={styles.input}>
            <CustomPickerSelect
              isError={!!validationErrors.subject.length}
              isBottomMargin={false}
              data={getSubjectData(feedbackSubject as AdditionalSubjectType)}
              value={subject}
              placeholder="Select Subject"
              onSelect={(item) => onChangeHandler(item?.value, 'subject')}
            />
            <ValidationErrorList errorList={validationErrors.subject} />
          </View>
          <View style={styles.input}>
            <TextInput
              value={email}
              onChangeText={(value) => onChangeHandler(value, 'email')}
              keyboardType="email-address"
              autoCapitalize="none"
              autoCorrect={false}
              placeholder="Email"
              isError={!!validationErrors.email.length}
              maxLength={AppConfig.maxTextInputLength}
              returnKeyType="next"
              ref={emailInputRef as TextInputRef}
              onSubmitEditing={() => {
                (nameInputRef as TextInputRef)?.current?.focusNextInput();
              }}
            />
            <ValidationErrorList errorList={validationErrors.email} />
          </View>
          <View style={styles.input}>
            <TextInput
              value={name}
              onChangeText={(value) => onChangeHandler(value, 'name')}
              style={styles.textInput}
              placeholder="Name (Optional)"
              isError={!!validationErrors.name.length}
              maxLength={AppConfig.maxTextInputLength}
              returnKeyType="next"
              ref={nameInputRef as TextInputRef}
              onSubmitEditing={() => {
                (phoneNumberInputRef as TextInputRef)?.current?.focusNextInput();
              }}
            />
            <ValidationErrorList errorList={validationErrors.name} />
          </View>
          <View style={styles.input}>
            <TextInput
              value={phoneNumber}
              onChangeText={(value) => onChangeHandler(value, 'phoneNumber')}
              placeholder="Phone Number (Optional)"
              isError={!!validationErrors.phoneNumber.length}
              maxLength={AppConfig.maxTextInputLength}
              returnKeyType="done"
              keyboardType="number-pad"
              ref={phoneNumberInputRef as TextInputRef}
              onSubmitEditing={() => {
                (messageInputRef as TextInputRef)?.current?.focusNextInput();
              }}
            />
            <ValidationErrorList errorList={validationErrors.phoneNumber} />
          </View>
          <View>
            <TextInput
              value={message}
              onChangeText={(value) => onChangeHandler(value, 'message')}
              numberOfLines={10}
              multiline
              style={styles.textInput}
              placeholder="Message"
              containerStyle={styles.textArea}
              isError={!!validationErrors.message.length}
              maxLength={textAreaLength}
              returnKeyType="next"
              ref={messageInputRef as TextInputRef}
            />
            <ValidationErrorList errorList={validationErrors.message} />
            <ValidationErrorList style={styles.validation} errorList={['Max 1000 Characters']} />
          </View>
          <Button disabled={isBtnDisabled} inverse={isBtnDisabled} title="SEND" onPress={onSendHandler} />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingVertical: Metrics.marginSection,
    paddingHorizontal: Metrics.baseMargin,
  },
  selectContainer: {
    paddingTop: 10,
  },
  input: {
    marginBottom: Metrics.halfMargin,
  },
  textInput: {
    height: '90%',
    textAlignVertical: 'top',
  },
  textArea: {
    height: 140,
  },
  validation: {
    marginBottom: Metrics.halfMargin,
  },
});

export default EmailUs;
