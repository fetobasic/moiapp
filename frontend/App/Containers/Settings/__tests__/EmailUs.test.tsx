import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import { yeti6GState } from 'App/Fixtures/yeti6GState';
import EmailUs from '../EmailUs';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('EmailUs', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
        canGoBack: jest.fn() as any,
      },
      route: {
        params: {
          device: yeti6GState,
          yeti: {
            connectedOk: true,
            thingName: '123',
            connectedAt: '2021-08-10T14:00:00.000Z',
          },
          firmwareVersionFailed: '1.0.0',
          thingName: '123',
          feedbackSubject: 'pairingProblem',
        },
      },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <EmailUs {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
