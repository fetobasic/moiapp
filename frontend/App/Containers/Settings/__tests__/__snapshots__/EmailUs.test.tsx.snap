// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailUs renders screen 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Email Us
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={false}
    keyboardShouldPersistTaps="handled"
    style={
      {
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View>
      <View
        style={
          {
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            {
              "marginBottom": 12,
            }
          }
        >
          <View
            style={
              {
                "position": "relative",
              }
            }
          >
            <View
              Icon="SvgMock"
              InputAccessoryView={[Function]}
              fixAndroidTouchableBug={true}
              items={
                [
                  {
                    "color": "rgba(191, 210, 43, 1)",
                    "label": "Pairing Problem",
                    "value": "pairingProblem",
                  },
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "label": "Report a Problem",
                    "value": "problem",
                  },
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "label": "Ask a Question",
                    "value": "question",
                  },
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "label": "Suggest an Improvement",
                    "value": "suggestion",
                  },
                ]
              }
              onClose={[Function]}
              onOpen={[Function]}
              onValueChange={[Function]}
              placeholder={
                {
                  "color": "rgba(147, 147, 147, 1)",
                  "inputLabel": "Select Subject",
                  "label": "None Selected",
                  "value": null,
                }
              }
              style={
                {
                  "iconContainer": {
                    "marginRight": 24,
                    "marginTop": 24,
                  },
                  "inputAndroid": {
                    "borderColor": "rgba(54, 54, 54, 1)",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginBottom": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 14,
                  },
                  "inputIOS": {
                    "borderColor": "rgba(54, 54, 54, 1)",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 17,
                    "marginBottom": 0,
                    "paddingHorizontal": 12,
                    "paddingVertical": 18,
                  },
                  "modalViewBottom": {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "borderLeftColor": "rgba(54, 54, 54, 1)",
                    "borderLeftWidth": 1,
                    "borderRightColor": "rgba(54, 54, 54, 1)",
                    "borderRightWidth": 1,
                    "marginHorizontal": 24,
                  },
                  "placeholder": {
                    "color": "rgba(255, 255, 255, 0.38)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 17,
                  },
                }
              }
              testID="customPicker"
              useNativeAndroidPickerStyle={false}
              value="pairingProblem"
            />
            <Text
              style={
                {
                  "backgroundColor": "rgba(27, 27, 27, 1)",
                  "color": "rgba(191, 210, 43, 1)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 12,
                  "fontWeight": "300",
                  "left": 8,
                  "lineHeight": 16,
                  "paddingHorizontal": 5,
                  "position": "absolute",
                  "top": -8,
                }
              }
            >
              Select Subject
            </Text>
          </View>
          <View />
        </View>
        <View
          style={
            {
              "marginBottom": 12,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "elevation": 12,
                "height": 56,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 12,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(54, 54, 54, 1)",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "color": "rgba(255, 255, 255, 0.871)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginLeft": 8,
                    "paddingHorizontal": 4,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Email
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                isError={false}
                keyboardAppearance="dark"
                keyboardType="email-address"
                maxFontSizeMultiplier={1.15}
                maxLength={50}
                numberOfLines={1}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onPressIn={[Function]}
                onSubmitEditing={[Function]}
                placeholder=""
                placeholderTextColor="rgba(255, 255, 255, 0.38)"
                returnKeyType="next"
                selectionColor="rgba(191, 210, 43, 1)"
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "paddingHorizontal": 12,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                value="testemail@example.com"
              />
            </View>
          </View>
          <View />
        </View>
        <View
          style={
            {
              "marginBottom": 12,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "elevation": 12,
                "height": 56,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 12,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(54, 54, 54, 1)",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "color": "rgba(255, 255, 255, 0.871)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginLeft": 8,
                    "paddingHorizontal": 4,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Name (Optional)
              </Text>
              <TextInput
                isError={false}
                keyboardAppearance="dark"
                maxFontSizeMultiplier={1.15}
                maxLength={50}
                numberOfLines={1}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onPressIn={[Function]}
                onSubmitEditing={[Function]}
                placeholder=""
                placeholderTextColor="rgba(255, 255, 255, 0.38)"
                returnKeyType="next"
                selectionColor="rgba(191, 210, 43, 1)"
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "paddingHorizontal": 12,
                    },
                    {
                      "height": "90%",
                      "textAlignVertical": "top",
                    },
                  ]
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                value=""
              />
            </View>
          </View>
          <View />
        </View>
        <View
          style={
            {
              "marginBottom": 12,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "elevation": 12,
                "height": 56,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 12,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(54, 54, 54, 1)",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "color": "rgba(255, 255, 255, 0.871)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginLeft": 8,
                    "paddingHorizontal": 4,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Phone Number (Optional)
              </Text>
              <TextInput
                isError={false}
                keyboardAppearance="dark"
                keyboardType="number-pad"
                maxFontSizeMultiplier={1.15}
                maxLength={50}
                numberOfLines={1}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onPressIn={[Function]}
                onSubmitEditing={[Function]}
                placeholder=""
                placeholderTextColor="rgba(255, 255, 255, 0.38)"
                returnKeyType="done"
                selectionColor="rgba(191, 210, 43, 1)"
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "paddingHorizontal": 12,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                value=""
              />
            </View>
          </View>
          <View />
        </View>
        <View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "elevation": 12,
                "height": 140,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 12,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(54, 54, 54, 1)",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "color": "rgba(255, 255, 255, 0.871)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginLeft": 8,
                    "paddingHorizontal": 4,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Message
              </Text>
              <TextInput
                isError={false}
                keyboardAppearance="dark"
                maxFontSizeMultiplier={1.15}
                maxLength={1000}
                multiline={true}
                numberOfLines={1}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onPressIn={[Function]}
                placeholder=""
                placeholderTextColor="rgba(255, 255, 255, 0.38)"
                returnKeyType="next"
                selectionColor="rgba(191, 210, 43, 1)"
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "paddingHorizontal": 12,
                    },
                    {
                      "height": "90%",
                      "textAlignVertical": "top",
                    },
                  ]
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                value=""
              />
            </View>
          </View>
          <View />
          <View
            style={
              {
                "marginBottom": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgba(255, 255, 255, 0.87)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 12,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "marginTop": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "rgba(255, 83, 29, 1)",
                  }
                }
              >
                *
              </Text>
               
              Max 1000 Characters
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "opacity": 1,
              },
              [
                {
                  "elevation": 12,
                  "shadowColor": "rgba(0, 0, 0, 1)",
                  "shadowOffset": {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 12,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              {
                "borderRadius": 24,
                "overflow": "hidden",
              }
            }
          >
            <RNSVGSvgView
              bbHeight="48"
              bbWidth="100%"
              focusable={false}
              height={48}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 48,
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGDefs>
                  <RNSVGLinearGradient
                    gradient={
                      [
                        0,
                        -13224394,
                        1,
                        1776411,
                      ]
                    }
                    gradientTransform={null}
                    gradientUnits={0}
                    name="grad"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0.5"
                  />
                </RNSVGDefs>
                <RNSVGRect
                  fill={
                    {
                      "brushRef": "grad",
                      "type": 1,
                    }
                  }
                  height="48"
                  propList={
                    [
                      "fill",
                    ]
                  }
                  width="100%"
                  x="0"
                  y="0"
                />
                <View
                  style={
                    [
                      {
                        "borderRadius": 24,
                        "marginHorizontal": 1,
                        "marginTop": 1,
                      },
                      {
                        "backgroundColor": "rgba(191, 210, 43, 1)",
                        "height": 46,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "paddingVertical": 12,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "Roboto-Medium",
                              "fontSize": 14,
                              "fontWeight": "700",
                              "lineHeight": 20,
                              "textTransform": "uppercase",
                            },
                            {
                              "color": "rgba(27, 27, 27, 1)",
                            },
                            undefined,
                          ]
                        }
                      >
                        SEND
                      </Text>
                    </View>
                  </View>
                </View>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
